src/atoms/Text/Text.native.js

```diff
@@ -0,0 +1,43 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components/native';
+import theme from '../../tokens/theme';
+import { Text as NativeText } from 'react-native';
+import { flattenArray } from '../../utils/index';
+
+const fontFamilies = {
+  light: `Lato-Regular`,
+  regular: `Lato-Regular`,
+  bold: `Lato-Bold`,
+};
+
+const Text = styled(NativeText)(
+  (props) => `
+  font-family: ${fontFamilies[props.weight]};
+  font-size: ${props.theme.fonts.size[props.size]};
+  letter-spacing: ${props.theme.fonts.letterSpacing[props.letterSpacing]};
+  color: ${props.color || props.theme.colors.shade[800]};
+  text-decoration-line: ${props.underline ? 'underline' : 'none'};
+  align-self: ${props.align};
+`,
+);
+
+Text.propTypes = {
+  children: PropTypes.node.isRequired,
+  weight: PropTypes.oneOf(Object.keys(theme.fonts.weight)),
+  size: PropTypes.oneOf(Object.keys(theme.fonts.size)),
+  letterSpacing: PropTypes.oneOf(Object.keys(theme.fonts.letterSpacing)),
+  color: PropTypes.oneOf(flattenArray(Object.values(theme.colors).map(Object.values))),
+  underline: PropTypes.bool,
+  align: PropTypes.oneOf(['flex-start', 'center', 'flex-end']),
+};
+
+Text.defaultProps = {
+  weight: 'regular',
+  size: 'm',
+  letterSpacing: 's',
+  underline: false,
+  align: 'flex-start',
+  testID: 'ds-text',
```

Comment:
for `testIds` we can basically create a file in `_helpers/automation-attributes` and export a function from there and use that in all the components.
[Something like this](https://github.com/auth0/cosmos/blob/776187d1fca7e61412903500125a5e9dc84ea7a2/core/components/_helpers/automation-attribute.ts) and see the usage [here](https://github.com/auth0/cosmos/blob/776187d1fc/core/components/atoms/text/text.tsx#L36)

Reply:
I'm not sure what would be the advantage of doing that?

@shubhnik test expert, opinion please!

Reply:
Lol no one becomes expert in writing 2 test cases 😆 
I had a concern which I raised in this PR here https://github.com/razorpay/blade/pull/17#discussion_r379844338